mypy/checkstrformat.py

```diff
@@ -449,6 +448,24 @@ def perform_special_format_checks(
                     call,
                     code=codes.STRING_FORMATTING,
                 )
+        a_type = get_proper_type(actual_type)
+        if isinstance(a_type, NoneType):
+            # Perform type check of alignment specifiers on None
+            if spec.format_spec and any(c in spec.format_spec for c in "<>^"):
+                specifierIndex = -1
+                for i in range(len("<>^")):
+                    if spec.format_spec[i] in "<>^":
+                        specifierIndex = i
```

Comment:
This logic seems confused -- why are we mixing `i` (refers to position in `<>^`) with indexing into `spec.format_spec`?

Reply:
This could be just 

```python
specifier_char = next(c for c in spec.format_spec if c in "<>^")
```

and then `specifier_char` instead of `spec.format_spec[specifierIndex]` (and next `if` is redundant, obviously, since you already check that such symbol exists). 

Aside, please stick to `snake_case` naming convention - it's both used in this codebase and recommended by PEP8, only very old stdlib parts still have some `camelCase` identifiers.